crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -62,9 +67,141 @@ impl Query {
     }
 
     // availableRange - pending impl. on IndexerV2
-    // dryRunTransactionBlock
     // coinMetadata
 
+    /// Simulate running a transaction to inspect its effects without
+    /// committing to them on-chain.
+    ///
+    /// `txBytes` either a `TransactionData` struct or a `TransactionKind`
+    ///     struct, BCS-encoded and then Base64-encoded.  The expected
+    ///     type is controlled by the presence or absence of `txMeta`: If
+    ///     present, `txBytes` is assumed to be a `TransactionKind`, if
+    ///     absent, then `TransactionData`.
+    ///
+    /// `txMeta` the data that is missing from a `TransactionKind` to make
+    ///     a `TransactionData` (sender address and gas information).  All
+    ///     its fields are nullable: `sender` defaults to `0x0`, if
+    ///     `gasObjects` is not present, or is an empty list, it is
+    ///     substituted with a mock Coin object, `gasPrice` defaults to
+    ///     the reference gas price, `gasBudget` defaults to the max gas budget
+    ///     and `gasSponsor` defaults to the sender.
+    ///
+    /// `skipChecks` optional flag to disable the usual verification
+    ///     checks that prevent access to objects that are owned by
+    ///     addresses other than the sender, and calling non-public,
+    ///     non-entry functions, and some other checks.  Defaults to false.
+    async fn dry_run_transaction_block(
+        &self,
+        ctx: &Context<'_>,
+        tx_bytes: String,
+        tx_meta: Option<TransactionMetadata>,
+        skip_checks: Option<bool>,
+    ) -> Result<DryRunResult> {
+        let skip_checks = skip_checks.unwrap_or(false);
+
+        let sui_sdk_client: &Option<SuiClient> = ctx
+            .data()
+            .map_err(|_| Error::Internal("Unable to fetch Sui SDK client".to_string()))
+            .extend()?;
+        let sui_sdk_client = sui_sdk_client
+            .as_ref()
+            .ok_or_else(|| Error::Internal("Sui SDK client not initialized".to_string()))
+            .extend()?;
+
+        let (sender_address, tx_kind, gas_price, gas_sponsor, gas_budget, gas_objects) =
+            if let Some(TransactionMetadata {
+                sender,
+                gas_price,
+                gas_objects,
+                gas_budget,
+                gas_sponsor,
+            }) = tx_meta
+            {
+                // This implies `TransactionKind`
+                let tx_kind = deserialize_tx_data::<TransactionKind>(&tx_bytes)?;
```

Comment:
Yeah, best to make this a private free function of this module, not a public function in `lib.rs`.